src/calendar-app.ts

```diff
@@ -0,0 +1,774 @@
+import type { CredentialsMigrationViewModel } from "./common/login/CredentialsMigrationViewModel.js"
+
+assertMainOrNodeBoot()
+bootFinished()
+
+const urlQueryParams = m.parseQueryString(location.search)
+
+assignEnvPlatformId(urlQueryParams)
+replaceNativeLogger(window, new Logger())
+
+let currentView: Component<unknown> | null = null
+window.tutao = {
+	client,
+	m,
+	lang,
+	root,
+	currentView,
+	locator: null,
+}
+
+client.init(navigator.userAgent, navigator.platform)
+
+if (!client.isSupported()) {
+	throw new Error("Unsupported")
+}
+
+// Setup exception handling after checking for client support, because in android the Error is caught by the unhandled rejection handler
+// and then the "Update WebView" message will never be show
+// we still want to do this ASAP so we can handle other errors
+setupExceptionHandling()
+
+// If the webapp is served under some folder e.g. /build we want to consider this our root
+const urlPrefixes = extractPathPrefixes()
+// Write it here for the WorkerClient so that it can load relative worker easily. Should do it here so that it doesn't break after HMR.
+window.tutao.appState = urlPrefixes
+
+const startRoute = getStartUrl(urlQueryParams)
+history.replaceState(null, "", urlPrefixes.prefix + startRoute)
+
+registerForMailto()
+
+import("./mail-app/translations/en")
+	.then((en) => lang.init(en.default))
+	.then(async () => {
+		await import("./common/gui/main-styles")
+
+		// do this after lang initialized
+		const { locator } = await import("./common/api/main/MainLocator")
+		await locator.init()
+
+		const { setupNavShortcuts } = await import("./common/misc/NavShortcuts.js")
+		setupNavShortcuts()
+
+		// this needs to stay after client.init
+		windowFacade.init(locator.logins)
+		if (isDesktop()) {
+			import("./common/native/main/UpdatePrompt.js").then(({ registerForUpdates }) => registerForUpdates(locator.desktopSettingsFacade))
+		}
+
+		const userLanguage = deviceConfig.getLanguage() && languages.find((l) => l.code === deviceConfig.getLanguage())
+
+		if (userLanguage) {
+			const language = {
+				code: userLanguage.code,
+				languageTag: languageCodeToTag(userLanguage.code),
+			}
+			lang.setLanguage(language).catch((e) => {
+				console.error("Failed to fetch translation: " + userLanguage.code, e)
+			})
+
+			if (isDesktop()) {
+				locator.desktopSettingsFacade.changeLanguage(language.code, language.languageTag)
+			}
+		}
+
+		locator.logins.addPostLoginAction(() => locator.postLoginActions())
+
+		if (isOfflineStorageAvailable()) {
+			const { CachePostLoginAction } = await import("./common/offline/CachePostLoginAction")
+			locator.logins.addPostLoginAction(
+				async () =>
+					new CachePostLoginAction(
+						await locator.calendarModel(),
+						locator.entityClient,
+						locator.progressTracker,
+						locator.cacheStorage,
+						locator.logins,
+					),
+			)
+		}
+
+		styles.init()
+
+		const contactViewResolver = makeViewResolver<
```

Comment:
do we need contact view resolver in calendar app?

Reply:
This is removed later